business_requirement/models/business_requirement.py

```diff
@@ -0,0 +1,260 @@
+        string='Master Project',
+        ondelete='set null',
+        copy=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    partner_id = fields.Many2one(
+        comodel_name='res.partner',
+        string='Customer',
+        store=True,
+        copy=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    sub_br_count = fields.Integer(
+        string='Count',
+        compute='_compute_sub_br_count'
+    )
+    priority = fields.Selection(
+        [('0', 'Low'), ('1', 'Normal'), ('2', 'High')],
+        required=True,
+        default='1'
+    )
+    requested_id = fields.Many2one(
+        'res.users',
+        string='Requested by',
+        required=True,
+        readonly=True,
+        default=lambda self: self.env.user,
+        states={
+            'draft': [('readonly', False)],
+            'confirmed': [('readonly', False)]}
+    )
+    confirmation_date = fields.Datetime(
+        copy=False,
+        readonly=True
+    )
+    confirmed_id = fields.Many2one(
+        'res.users', string='Confirmed by',
+        copy=False,
+        readonly=True
+    )
+    reviewed_date = fields.Datetime(
+        copy=False,
+        readonly=True
+    )
+    reviewed_id = fields.Many2one(
+        'res.users',
+        string='Reviewed by',
+        copy=False,
+        readonly=True
+    )
+    approval_date = fields.Datetime(
+        copy=False,
+        readonly=True
+    )
+    approved_id = fields.Many2one(
+        'res.users',
+        string='Approved by',
+        copy=False,
+        readonly=True
+    )
+    company_id = fields.Many2one(
+        'res.company', string='Company',
+        required=True, readonly=True, states={'draft': [('readonly', False)]},
+        default=lambda self: self.env.user.company_id,
+    )
+
+    @api.multi
+    @api.onchange('project_id')
+    def project_id_change(self):
+        for br in self:
+            if br.project_id and br.project_id.partner_id:
+                br.partner_id = br.project_id.partner_id
+
+    @api.model
+    def create(self, vals):
+        if vals.get('name', '/') == '/':
+            vals['name'] = self.env['ir.sequence'].\
+                next_by_code('business.requirement')
+        return super(BusinessRequirement, self).create(vals)
+
+    @api.multi
+    @api.depends('parent_id')
+    def _get_level(self):
+        for br in self:
+            level = br.parent_id and br.parent_id.level + 1 or 1
+            br.level = level
+
+    @api.multi
+    @api.depends('business_requirement_ids')
+    def _compute_sub_br_count(self):
+        for br in self:
+            br.sub_br_count = len(br.business_requirement_ids)
+
+    @api.model
+    def _get_states(self):
+        states = [
+            ('draft', 'Draft'),
+            ('confirmed', 'Confirmed'),
+            ('approved', 'Approved'),
+            ('stakeholder_approval', 'Stakeholder Approval'),
```

Comment:
@elicoidal in V8 the states values have `stakeholder_approval`, we should be careful when changing this item. It also implies to create migration scripts.

Maybe changing only the string 'Stakeholder Approved' would be enough?